src/main/java/org/cyclops/integrateddynamics/core/evaluate/operator/Operators.java

```diff
@@ -3341,6 +3341,86 @@ public NBTTagCompound getOutput(Triple<NBTTagCompound, String, OperatorBase.Safe
                 return ValueObjectTypeRecipe.ValueRecipe.of(null);
             }).build());
 
+    /**
+     * ------------------------------------ PARSE OPERATORS ------------------------------------
+     */
+
+    /**
+     * Boolean Parse operator which takes a string of form `/(F(alse)?|[+-]?(0x|#)?0+|)/i`.
+     */
+    public static final IOperator PARSE_BOOLEAN = Operators.REGISTRY.register(new ParseOperator<>(ValueTypes.BOOLEAN, v -> {
+      ValueTypeString.ValueString value = v.getValue(0);
+      java.util.regex.Pattern p = java.util.regex.Pattern.compile("\\A(F(alse)?|[+-]?(0x|#)?0+|)\\z", java.util.regex.Pattern.CASE_INSENSITIVE);
+      if (p.matcher(value.getRawValue().trim()).matches()){
+        return ValueTypeBoolean.ValueBoolean.of(false);
+      } else {
+        return ValueTypeBoolean.ValueBoolean.of(true);
+      }
+    }));
+
+    /**
+     * Double Parse operator which takes a string of a form Double.parseDouble() can consume,
+     * `/([+-]?)(Inf(inity)?|\u221E)/i`, or Long.decode() can consume.
+     */
+    public static final IOperator PARSE_DOUBLE = Operators.REGISTRY.register(new ParseOperator<>(ValueTypes.DOUBLE, v -> {
+      ValueTypeString.ValueString value = v.getValue(0);
+      try {
+        return ValueTypeDouble.ValueDouble.of(Double.parseDouble(value.getRawValue()));
+      } catch (NumberFormatException e) {
+        try {
+          // \u221E = infinity symbol
+          java.util.regex.Pattern p = java.util.regex.Pattern.compile("\\A([+-]?)(Inf(inity)?|\u221E)\\z", java.util.regex.Pattern.CASE_INSENSITIVE);
+          java.util.regex.Matcher m = p.matcher(value.getRawValue().trim());
+          if (m.matches()){
+            if (m.group(1).equals("-")){
+              return ValueTypeDouble.ValueDouble.of(Double.NEGATIVE_INFINITY);
+            }
+            return ValueTypeDouble.ValueDouble.of(Double.POSITIVE_INFINITY);
+          }
+          // Try as a long
+          return ValueTypeDouble.ValueDouble.of((double) Long.decode(value.getRawValue()));
+        } catch (NumberFormatException e2) {
+          throw new EvaluationException("'" + value.getRawValue() + "' is not parsable as a '" + ValueTypes.DOUBLE.getTranslationKey() + "'");
```

Comment:
I would not use `getTranslationKey` in error messages, as this will not be localized, so it would not really be readable for players.

Reply:
I'd just use a plain string :-)